### ISSUE TITLE
GTK version of the application is broken

### ISSUE THREAD

**Issue by Kazark** [PolyglotSymposium/void#5]
At some point I seem to have broken the GTK version of the application without realizing it. When attempting to run it in GTK2 on my machine, I get this:

```

(Void.Gtk2:16638): Gdk-CRITICAL **: IA__gdk_cairo_create: assertion 'GDK_IS_DRAWABLE (drawable)' failed
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) Cairo.NativeMethods.cairo_rectangle (intptr,double,double,double,double) <IL 0x00014, 0xffffffff>
  at Cairo.Context.Rectangle (double,double,double,double) <IL 0x0000b, 0x00043>
  at Eto.GtkSharp.Drawing.GraphicsHandler.FillRectangle (Eto.Drawing.Brush,single,single,single,single) <IL 0x00017, 0x00073>
  at Eto.Drawing.Graphics.FillRectangle (Eto.Drawing.Color,Eto.Drawing.RectangleF) <IL 0x0002a, 0x001dd>
  at Void.EtoArtist.RenderBlock (Void.ViewModel.ScreenBlockObject) [0x0001d] in /home/keith/code/void/Void/Void/EtoArtist.cs:45
  at Void.EtoArtist.Draw (Void.ViewModel.DrawingObject) [0x00050] in /home/keith/code/void/Void/Void/EtoArtist.cs:29
  at <StartupCode$Void-ViewModel>.$VoidController/initializeView@23.Invoke (System.Action`1<Void.ViewModel.DrawingObject>) [0x0001c] in /home/keith/code/void/Void/Void.ViewModel/VoidController.fs:23
  at Void.MainForm/<SubscribeToDraw>c__AnonStorey1.<>m__0 (object,Eto.Forms.PaintEventArgs) [0x00028] in /home/keith/code/void/Void/Void/MainForm.cs:69
  at Eto.Forms.Drawable.OnPaint (Eto.Forms.PaintEventArgs) <IL 0x00013, 0x0002f>
  at Eto.Forms.Drawable/Callback/<OnPaint>c__AnonStorey0.<>m__0 () <IL 0x0000c, 0x0001e>
  at Eto.Platform.Invoke (System.Action) <IL 0x00008, 0x0002e>
  at Eto.Forms.Drawable/Callback.OnPaint (Eto.Forms.Drawable,Eto.Forms.PaintEventArgs) <IL 0x0002b, 0x00137>
  at Eto.GtkSharp.Forms.Controls.DrawableHandler.Update (Eto.Drawing.Rectangle) <IL 0x00035, 0x000f2>
  at Eto.Forms.Drawable.Update (Eto.Drawing.Rectangle) <IL 0x00007, 0x000b5>
  at Void.MainForm.TriggerDraw () [0x0000d] in /home/keith/code/void/Void/Void/MainForm.cs:75
  at Void.ViewModel.ViewController.handleCommand (Void.Core.Command) [0x00041] in /home/keith/code/void/Void/Void.ViewModel/VoidController.fs:40
  at Void.ViewModel.MainController.handleCommand (Void.Core.Command) [0x000d5] in /home/keith/code/void/Void/Void.ViewModel/VoidController.fs:93
  at Void.ViewModel.MainController.handleCommand (Void.Core.Command) [0x00086] in /home/keith/code/void/Void/Void.ViewModel/VoidController.fs:90
  at Void.ViewModel.MainController.initializeVoid () [0x00033] in /home/keith/code/void/Void/Void.ViewModel/VoidController.fs:75
  at Void.MainForm..ctor () [0x00042] in /home/keith/code/void/Void/Void/MainForm.cs:21
  at Void.Gtk2.Program.Main (string[]) [0x0000b] in /home/keith/code/void/Void/Void.Gtk2/Program.cs:11
  at (wrapper runtime-invoke) <Module>.runtime_invoke_void_object (object,intptr,intptr,intptr) <IL 0x00050, 0xffffffff>

Native stacktrace:

    /usr/bin/mono() [0x4accac]
    /usr/bin/mono() [0x50451f]
    /usr/bin/mono() [0x42a7c7]
    /lib/x86_64-linux-gnu/libpthread.so.0(+0x10340) [0x7f7eb8844340]
    /usr/lib/x86_64-linux-gnu/libcairo.so.2(cairo_rectangle+0x1) [0x7f7eaeee37b1]
    [0x418e7192]

Debug info from gdb:

Could not attach to process.  If your uid matches the uid of the target
process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try
again as the root user.  For more details, see /etc/sysctl.d/10-ptrace.conf
ptrace: Operation not permitted.
No threads.

=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================
```

When trying to run GTK3, it does not work either, but if I remember correctly, it never worked because I don't have GTK3 on my machine or something.

Is this a bug in Eto? I am starting to wonder whether it's a good idea to use Eto. Because this isn't a useful forms/controls GUI, but largely drawing-based, maybe it would just make sense to implement each GUI by hand. The documentation and support would be better, but we would have to maintain multiple things explicitly. What do you think, @mjgpy3?

Also, this shows the need for somehow getting a launch of the application into the build process, so I do not have to be continually smoke checking it manually. The build itself is not broken and yet the app dies when it attempts to launch. That seems like a problem which should have CI on it because this happened to me before.

**Comment by mjgpy3** [comment]
Frankly, because you're finding these issues in Eto (and we can prove they are in fact Eto issues) this regularly and this early in the project, there are two possible paths I suggest:
1. We drop it (I mean, are you really hitting edge-cases in the framework or are these conditions trivial?) or
2. Try to fix Eto (https://github.com/picoe/Eto)

Thoughts?

**Comment by Kazark** [comment]
The open source guy in me would like to fix Eto. Maybe we should at least submit a couple of issues.

On the other hand, given the nature of this application, it really might make sense not to use Eto. I think we should create a branch or branches on which to explore other possibilities.

**Comment by mjgpy3** [comment]
:+1:

**Comment by Kazark** [comment]
Locally I started a WinForms branch because I think we may be able to run a WinForms app on Mono on Linux and because Eto seemed to be designed similarly to WinForms. So if you are going to play around with a GUI library you might try GTK or WPF or something else.

**Comment by Kazark** [comment]
The WinForms experiment is going well, as you can see on the [feature/winforms](https://github.com/PolyglotSymposium/void/tree/feature/winforms) branch. It resolved the `;` issue (#4) and it has let me to greater clarity because of better documentation and because IntelliSense works (and just because it made me think through some things and understand the UI a little better). I still need to test and see if it will resolve this issue and work on Linux.

**Comment by Kazark** [comment]
This is fixed by the feature/winforms branch! It builds and runs on Linux no problem! It doesn't look quite as nice as it did when the GTK was working, but honestly, a little love on the fonts at some point and that should be no big deal. WinForms is looking promising! If I can iron out a few more things, I'll submit a pull request for it.

**Comment by Kazark** [comment]
This is closed by #12. There is no longer a GTK version of the app.
